testing/jormungandr-testing-utils/src/testing/vit/vote_cast_register.rs

```diff
@@ -0,0 +1,527 @@
+#![allow(dead_code)]
+
+use chain_impl_mockchain::certificate::VotePlanId;
+use std::collections::HashMap;
+use std::iter;
+use std::ops::Range;
+use thiserror::Error;
+
+pub struct VoteCastCounter {
+    register: Vec<WalletVoteCastPosition>,
+}
+
+impl VoteCastCounter {
+    pub fn new(wallet_count: usize, vote_plans: Vec<(VotePlanId, u8)>) -> Self {
+        Self {
+            register: iter::from_fn(|| Some(WalletVoteCastPosition::new(vote_plans.clone())))
+                .take(wallet_count)
+                .collect(),
+        }
+    }
+
+    pub fn available_to_send(&self) -> usize {
+        self.register.iter().map(|x| x.available_to_send()).sum()
+    }
+
+    pub fn is_drained(&self) -> bool {
+        self.available_to_send() == 0
+    }
+
+    pub fn advance_single(&mut self, wallet_idx: usize) -> Result<WalletVotesToCast, Error> {
+        self.advance_batch(1, wallet_idx)
+    }
+
+    pub fn advance_batch(
+        &mut self,
+        mut requested_batch_size: usize,
+        wallet_idx: usize,
+    ) -> Result<WalletVotesToCast, Error> {
+        let mut wallet_votes_to_cast = WalletVotesToCast::new();
+
+        let vote_plan = self.register.get_mut(wallet_idx).unwrap();
+
+        let votes_to_cast = vote_plan.advance_batch(requested_batch_size).map_err(|_| {
+            Error::NoMoreRequestsToSentForLoad {
+                wallet_idx,
+                requested_batch_size,
+            }
+        })?;
+        votes_to_cast.iter().for_each(|votes_to_cast| {
+            requested_batch_size -= votes_to_cast.range.end - votes_to_cast.range.start
+        });
+        wallet_votes_to_cast.add(wallet_idx, votes_to_cast);
+
+        Ok(wallet_votes_to_cast)
+    }
+}
+
+pub struct WalletVoteCastPosition {
+    register: Vec<VotePlanVoteCastPosition>,
+}
+
+impl WalletVoteCastPosition {
+    pub fn new(vote_plans: Vec<(VotePlanId, u8)>) -> Self {
+        Self {
+            register: vote_plans
+                .iter()
+                .map(|(id, limit)| VotePlanVoteCastPosition::new(id.clone(), *limit))
+                .collect(),
+        }
+    }
+
+    pub fn has_next(&self) -> bool {
```

Comment:
nit: this sounds like I'm reading a java iterator implementation

Reply:
I guess it is not good :) removing then